dashboard/app/models/census/state_cs_offering.rb

```diff
@@ -85,7 +99,23 @@ def self.infer_no(state_code)
     INFERRED_NO_EXCLUSION_LIST.exclude? state_code.upcase
   end
 
-  def self.construct_state_school_id(state_code, row_hash)
+  def self.construct_state_school_id(state_code, row_hash, school_year)
+    # Using V2 format.
+    if state_uses_format_v2(state_code, school_year)
+      nces_id = row_hash['nces_id']
+      state_school_id = row_hash['state_school_id']
+
+      # The V2 format requires either nces_id or state_school_id.
+      if nces_id != UNSPECIFIED_VALUE
+        return School.find_by(id: nces_id)&.state_school_id
```

Comment:
If `find_by` the NCES ID does not return a record, this will return `nil`.  Could we try to find the record and then fallback to returning the `state_school_id` provided in the CSV?

Reply:
Good call!